src/homeWork5/home_work5.ts

```diff
@@ -0,0 +1,80 @@
+//Створіть класи Circle, Rectangle, Square і Triangle. У кожного з них є загальнодоступний метод calculateArea.
+// У кожної фігури є загальнодоступні властивості - колір і назва, які не можна змінювати після створення.
+//У Square і Rectangle зі свого боку є ще додатковий метод print, який виводить рядок із формулою розрахунку площі.
+
+class Shape {
+  //базовий клас який має захищенні поля та базовий метод розрахунку calculateArea().
+  protected readonly color: string;
+  protected readonly name: string;
+
+  constructor(color: string, name: string) {
+    this.color = color;
+    this.name = name;
+  }
+
+  public calculateArea(): number {
```

Comment:
This approach does not allow you to strict that method for children's classes. Try an abstract class.